src/test/java/spoon/test/fieldaccesses/FieldAccessTest.java

```diff
@@ -388,7 +389,7 @@ public void testFieldAccessOnUnknownType() {
 			@Override
 			public <T> void visitCtFieldWrite(CtFieldWrite<T> fieldWrite) {
 				visited++;
-				assertEquals("array", ((CtVariableWrite) fieldWrite.getTarget()).getVariable().getSimpleName());
+				assertEquals("array", ((CtVariableAccess) fieldWrite.getTarget()).getVariable().getSimpleName());
```

Comment:
Is this related to that PR?

Reply:
That's actually really since we are inside a `visitCtFieldWrite`. So it would mean the target of a field write is something we cannot write?

Reply:
I propose that you split in two PR: I'm ok to merge everything, but on this one we really should discuss and understand what happens.

Reply:
Sure. No problem.